### PR TITLE
Add new modal component

### PR DESCRIPTION
**Pull request by NipuniBhagya**
Modal component preview
image

### CODE REVIEW

File: apps/user-portal/src/components/shared/modal.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { MouseEvent } from "react";
+import { Button, Icon, Modal } from "semantic-ui-react";
+
+interface ModalProps {
```
Comment (brionmario) [inline apps/user-portal/src/components/shared/modal.tsx]: Can't we extent this interface with `ModalProps` from `semantic` and just put extra attributes like `type` etc.

File: apps/user-portal/src/components/shared/modal.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        }
+    };
+
+    return (
+        <Modal
+            size="mini"
+            className="custom-modal"
+            dimmer="blurring"
+            open={ open }
+            onClose={ onClose }
```
Comment (brionmario) [inline apps/user-portal/src/components/shared/modal.tsx]: we better remove size and dimmer from here and spread the unhandled props. If the modal size & dimmer has to be `mini` & `blurring` by default, let's add them in defaultProps.

File: modules/theme/src/themes/default/elements/button.variables [hunk 3]
```diff
@@ -33,3 +33,14 @@
...
+/*--------------------
+ Modal primary button
+----------------------*/
+
+@positivePrimaryButtonColor: @positiveColor;
+@negativePrimaryButtonColor: @negativeColor;
+@warningPrimaryButtonColor: @warningColor;
+@infoPrimaryButtonColor: @infoColor;
+
+@modalPrimaryButtonFontColor: #FFFFFF;
```
Comment (brionmario) [inline modules/theme/src/themes/default/elements/button.variables]: We better put `@white` here.

File: modules/theme/src/themes/default/modules/modal.overrides [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    stroke:#7CB342;
+}
+
+.red-stroke {
+    stroke: #FF6245;
+}
+
+.yellow-stroke {
+    stroke: #FFC107;
+}
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/modules/modal.overrides]: Shall we scope these styles with `svg-box`? And shall we use the `negative, warning` ... etc terms?

File: modules/theme/src/themes/default/modules/modal.overrides [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+.circular circle.path {
+    stroke-dasharray: 330;
+    stroke-dashoffset: 0;
+    stroke-linecap: round;
+    opacity: 0.4;
+    animation: 0.7s draw-circle ease-out;
+}
+
+/*------- Checkmark ---------*/
+.checkmark{
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/modules/modal.overrides]:
```suggestion
.checkmark {
```

File: modules/theme/src/themes/default/modules/modal.overrides [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+	stroke-width: 6.25;
+    stroke-linecap: round;
+	position:absolute;
+    top: 56px;
+    left: 49px;
+    width: 52px;
+    height: 40px;
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/modules/modal.overrides]: Fix indentation

File: modules/theme/src/themes/default/modules/modal.overrides [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    top: 40px;
+    left: 68px;
+    width: 15px;
+    height: 70px;
+    animation: 0.5s alert-sign-bounce cubic-bezier(0.175, 0.885, 0.32, 1.275);
+}
+
+.warning-icon .dot {
+    stroke:none;
+    fill: #FFC107;
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/modules/modal.overrides]: Move this to a variable

**Pull request merged**